applications/Wasm Smart Contract Bounty (Ink!).md

```diff
@@ -0,0 +1,365 @@
+#  Wasm Smart Contract Bounty (Ink!)
+
+- **Team Name:** Wasm Smart Contract Bounty (Ink!)
+- **Payment Address:** 13s3c37rwrZfa8WQcZPdqE1zi1hGAguf2TWfWMr8f5aLBrg
+
+
+## Project Overview :page_facing_up:
+### Overview
+The project goal is to build the trade pairs liquidity system for Astar network in order to implement the Uniswap-like functionality on !ink. The idea of the project came from https://github.com/use-inkubator/Ecosystem-Grants/blob/master/README.md , as a Canary DApp showcase in order to enrich the !ink Ecosystem. 
+
+
+### Project Details
+
+We've prepared simple RUST interfaces, please see them below: 
+
+```rs
+type Balance = <ink::env::DefaultEnvironment as ink::env::Environment>::Balance;
+type AccountId = <ink::env::DefaultEnvironment as ink::env::Environment>::AccountId;
+type Hash = <ink::env::DefaultEnvironment as ink::env::Environment>::Hash;
+​
+/// The ERC-20 error types.
+#[derive(Debug, PartialEq, Eq, scale::Encode, scale::Decode)]
+#[cfg_attr(feature = "std", derive(scale_info::TypeInfo))]
+pub enum Error {
+    /// Returned if not enough balance to fulfill a request is available.
+    InsufficientBalance,
+    /// Returned if not enough allowance to fulfill a request is available.
+    InsufficientAllowance,
+}
+​
+#[ink::trait_definition]
+pub trait ERC20 {
+    #[ink(message)]
+    fn name(&self) -> String;
+​
+    #[ink(message)]
+    fn symbol(&self) -> String;
+​
+    #[ink(message)]
+    fn decimals(&self) -> u8;
+​
+    #[ink(message)]
+    fn total_supply(&self) -> Balance;
+​
+    #[ink(message)]
+    fn balance_of(&self, owner: AccountId) -> Balance;
+​
+    #[ink(message)]
+    fn allowance(&self, owner: AccountId, spender: AccountId) -> Balance;
+​
+    #[ink(message)]
+    fn transfer(&mut self, to: AccountId, value: Balance) -> Result<(), ()>;
+​
+    #[ink(message)]
+    fn approve(&mut self, spender: AccountId, value: Balance) -> Result<(), ()>;
+​
+    #[ink(message)]
+    fn transfer_from(&mut self, from: AccountId, to: AccountId, value: Balance) -> Result<(), ()>;
+}
+​
+#[ink::trait_definition]
+pub trait InkswapERC20: ERC20 {
+    #[ink(message)]
+    fn domain_separator(&self) -> Hash;
+​
+    #[ink(message)]
+    fn permit_typehash(&self) -> Hash;
+​
+    #[ink(message)]
+    fn nonces(&self, owner: AccountId) -> u128;
+​
+    #[ink(message)]
+    fn permit(
+        &mut self,
+        owner: AccountId,
+        address: AccountId,
+        value: Balance,
+        deadline: u128,
+        v: u8,
+        r: Hash,
+        s: Hash,
+    ) -> Result<()>;
+}
+​
+#[ink::trait_definition]
+pub trait InkswapERC20: InkswapERC20 {
```

Comment:
Some typo.

Reply:
yes, agree with Michael 
Please take a look on this grant application - https://github.com/Chidasan/Ecosystem-Grants/blob/3be440fbb272358a4c7d3d0ee0539f4529bb0961/applications/starlay_finance.md
or others